packages/blade/src/components/Input/BaseInput/StyledBaseInput.native.tsx

```diff
@@ -64,35 +72,48 @@ const KeyboardTypeToNativeValuesMap = {
   decimal: 'decimal-pad',
 };
 
-const StyledNativeBaseInput = styled.TextInput<
-  Omit<
-    StyledBaseInputProps,
-    'accessibilityProps' | 'setCurrentInteraction' | 'currentInteraction'
-  > & {
-    isTextArea?: boolean;
-    isFocused: boolean;
-    autoCompleteType?: typeof autoCompleteSuggestionTypeAndroid[keyof typeof autoCompleteSuggestionTypeAndroid];
-  }
->((props) => ({
-  ...getBaseInputStyles({
-    theme: props.theme,
-    isFocused: props.isFocused,
-    isDisabled: !props.editable,
-    validationState: props.validationState,
-    leadingIcon: props.leadingIcon,
-    prefix: props.prefix,
-    interactionElement: props.interactionElement,
-    suffix: props.suffix,
-    trailingIcon: props.trailingIcon,
-  }),
-  lineHeight: undefined,
-  textAlignVertical: 'top',
-  height: props.isTextArea
-    ? `${props.theme.typography.lineHeights.xl * (props.numberOfLines ?? 0)}px`
-    : '36px',
-}));
+type StyledComponentInputProps = Omit<
+  StyledBaseInputProps,
+  'accessibilityProps' | 'setCurrentInteraction' | 'currentInteraction'
+> & {
+  isTextArea?: boolean;
+  isFocused: boolean;
+  autoCompleteType?: typeof autoCompleteSuggestionTypeAndroid[keyof typeof autoCompleteSuggestionTypeAndroid];
+  onPress?: (event: GestureResponderEvent) => void;
+};
+
+const getRNInputStyles = (
+  props: StyledComponentInputProps &
+    ThemeProps<DefaultTheme> &
+    (TextInputProps | TouchableHighlightProps),
+): CSSObject => {
+  return {
+    ...getBaseInputStyles({
+      theme: props.theme,
+      isFocused: props.isFocused,
+      isDisabled: 'editable' in props ? !props.editable : undefined,
```

Comment:
🤔 Whats happening here? why editable is not in `props`? 

It should just be `!props.editable` 
And add editable in here: 

```
type StyledComponentInputProps = {
  editable: TextInputProps['editable'];
}
```

Reply:
ops right yes fixed it